packages/teams-js/src/internal/videoUtils.ts

```diff
@@ -108,6 +141,142 @@ function createProcessedStreamGenerator(
   return generator;
 }
 
+/**
+ * The function to create a processed video track from the original video track.
+ * It reads frames from the video track and pipes them to the video frame callback to process the frames.
+ * The processed frames are then enqueued to the generator.
+ * The generator can be registered back to the media stream so that the host can get the processed frames.
+ */
+function createProcessedStreamGeneratorWithMetadata(
```

Comment:
There are some repeated code in this function and `createProcessedStreamGenerator`, try to make them reusable.

Reply:
refactored